core/src/main/java/org/apache/brooklyn/core/sensor/ReleaseableLatch.java

```diff
@@ -56,8 +60,11 @@
         }
 
         private static class MaxConcurrencyLatch implements ReleaseableLatch {
+            private static final Logger LOG = LoggerFactory.getLogger(MaxConcurrencyLatch.class);
+
             private int permits;
             private transient final Semaphore sem;
+            private transient final Set<Entity> ownerEntities = new ConcurrentHashSet<Entity>();
```

Comment:
The `transient` makes me think you're trying to handle persistence of the `MaxConcurrencyLatch` - is that right?

If so, then this will fail after rebind because the field `sem` will be null. We'd need a readResolve method to set it, or a getter that lazily instantiates the semaphore.